lib/ddtrace/contrib/active_storage/events/delete.rb

```diff
@@ -0,0 +1,59 @@
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/contrib/active_storage/event'
+require 'ddtrace/ext/http'
+require 'ddtrace/contrib/active_storage/ext'
+
+module Datadog
+  module Contrib
+    module ActiveStorage
+      module Events
+        # Defines instrumentation for 'service_delete.active_storage' event.
+        # From: https://edgeguides.rubyonrails.org/active_support_instrumentation.html#active-storage
+        module Delete
+          include ActiveStorage::Event
+
+          EVENT_NAME = 'service_delete.active_storage'.freeze
+
+          module_function
+
+          def event_name
+            self::EVENT_NAME
+          end
+
+          def span_name
+            Ext::SPAN_ACTION
+          end
+
+          def span_type
+            # Interacting with a cloud based image service
+            Datadog::Ext::HTTP::TYPE_OUTBOUND
+          end
+
+          def resource_prefix
+            Ext::ACTION_DELETE
+          end
+
+          def process(span, _event, _id, payload)
+            as_key = payload[:key]
+            as_service = payload[:service]
+
+            span.service = configuration[:service_name]
+            span.resource = "#{resource_prefix} #{as_service}"
+            span.span_type = span_type
+
+            # Set analytics sample rate
+            if Contrib::Analytics.enabled?(configuration[:analytics_enabled])
+              Contrib::Analytics.set_sample_rate(span, configuration[:analytics_sample_rate])
+            end
+
+            # Measure service stats
+            Contrib::Analytics.set_measured(span)
+
+            span.set_tag(Ext::TAG_SERVICE, as_service)
+            span.set_tag(Ext::TAG_KEY, as_key)
```

Comment:
The events are not all the same, but I see quite a few similarities between them.
Would it be possible to create a shared component for these?
I understand that not all the fields are the same, and would require custom treatment.